nfs/src/rpc_task.cpp

```diff
@@ -1659,6 +1659,17 @@ void rpc_task::run_read()
             assert(child_tsk->rpc_api->read_task.get_size() <=
                    rpc_api->read_task.get_size());
 
+            /*
+             * Now we are going to call read_from_server() which will issue
+             * an NFS read that will read the data from the NFS server and
+             * update the buffer. Grab the membuf lock, this will be
+             * unlocked in read_callback() once the data has been
+             * written to the buffer and it's marked uptodate.
+             *
+             * Note: This will block till the lock is obtained.
+             */
+            bc_vec[i].get_membuf()->set_locked();
```

Comment:
anytime we grab the lock there's a possibility that the buffer got updated by someone else, so we need to do the is_uptodate check as usual